core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -69,8 +70,8 @@ public UnsafeSorterSpillReader(
       bufferSizeBytes = DEFAULT_BUFFER_SIZE_BYTES;
     }
 
-    final BufferedInputStream bs =
-        new BufferedInputStream(new FileInputStream(file), (int) bufferSizeBytes);
+    final NioBasedBufferedFileInputStream bs =
```

Comment:
Yes, there are other instance in the code where we are using `BufferedInputStream`. I will file raise another PR to replace those with `NioBasedBufferedFileInputStream`.